WeakAuras/Prototypes.lua

```diff
@@ -984,8 +984,8 @@ end
 ---@return boolean result
 function WeakAuras.ValidateNumericOrPercent(info, val)
   if val ~= nil and val ~= "" then
-    local percent = string.match(val, "(%d+)%%")
-    local number = percent and tonumber(percent) or tonumber(val)
+    local percent, count = val:gsub("%%", " ", 1)
+    local number = count == 1 and tonumber(percent) or tonumber(val)
```

Comment:
According to your information, and I haven't tested it:
"%12" would be incorrectly parsed with your code?

Meaning the right thing to do is to search for the first %, and do a tonumber on the substring before that.

Reply:
Good catch, yeah

Reply:
> Search for the first %, and do a tonumber on the substring before that.

This would fix the case of "%12" but break "12% abcd"

The handling was fixed to find the last % that is followed only by spaces and end of string - regex `"%% *$"`. Then doing tonumber on the substring before that.